app/src/main/java/com/infomaniak/swisstransfer/ui/screen/onboarding/OnboardingScreen.kt

```diff
@@ -0,0 +1,195 @@
+    val coroutineScope = rememberCoroutineScope()
+
+    val isHighlighted = Page.entries.associateWith { rememberSaveable { mutableStateOf(false) } }
+
+    // Start the highlighting of the text when the associated page is reached in the HorizontalPager
+    LaunchedEffect(pagerState.currentPage) {
+        val currentPage = Page.entries[pagerState.currentPage]
+        val setIsHighlighted = isHighlighted[currentPage]?.component2()
+        setIsHighlighted?.invoke(true)
+    }
+
+    BackHandler(pagerState.currentPage > 0) {
+        coroutineScope.launch {
+            pagerState.animateScrollToPage(pagerState.currentPage - 1)
+        }
+    }
+
+    val onboardingPages = buildList {
+        Page.entries.forEach { page ->
+            add(page.toOnboardingPage(isHighlighted))
+        }
+    }
+
+    OnboardingScaffold(
+        pagerState = pagerState,
+        onboardingPages = onboardingPages,
+        bottomContent = {
+            BottomContent(
+                modifier = Modifier
+                    .padding(it)
+                    .consumeWindowInsets(it),
+                isLastPage = { isLastPage },
+                startMainActivity = goToMainActivity,
+                goToNextPage = { coroutineScope.launch { pagerState.animateScrollToPage(pagerState.currentPage + 1) } },
+            )
+        },
+    )
+}
+
+@Composable
+private fun Page.toOnboardingPage(isHighlighted: Map<Page, MutableState<Boolean>>) = OnboardingPage(
+    background = background.image(),
+    illustration = { Illustration(illustration) },
+    text = {
+        TitleAndDescription(
+            page = this,
+            isHighlighted = { isHighlighted[this]?.value ?: false }
+        )
+    }
+)
+
+@Composable
+private fun Illustration(illustration: ThemedImage) {
+    Image(illustration.image(), contentDescription = null)
+}
+
+@Composable
+private fun TitleAndDescription(page: Page, isHighlighted: () -> Boolean) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally) {
+        Text(
+            textAlign = TextAlign.Center,
+            text = stringResource(page.titleRes),
+            style = SwissTransferTheme.typography.specificLight22,
+        )
+
+        Spacer(modifier = Modifier.height(Margin.Mini))
+
+        HighlightedText(
+            modifier = Modifier.widthIn(max = Dimens.DescriptionWidth),
+            templateRes = page.subtitleTemplateRes,
+            argumentRes = page.subtitleArgumentRes,
+            style = SwissTransferTheme.typography.h1,
+            angleDegrees = page.highlightAngleDegree,
+            isHighlighted = isHighlighted,
+        )
+    }
+}
+
+@Composable
+private fun BottomContent(
+    modifier: Modifier = Modifier,
+    isLastPage: () -> Boolean,
+    startMainActivity: () -> Unit,
+    goToNextPage: () -> Unit
+) {
+    Box(
+        contentAlignment = Alignment.Center,
+        modifier = modifier
+            .fillMaxWidth()
+            .height(140.dp),
```

Comment:
Maybe put this magic value as a private const

Reply:
I don't really know if we need to name all dimensions that are only used once